peps/pep-2026.rst

```diff
@@ -0,0 +1,570 @@
+This introduces additional complexity and cost.
+
+CPython changes
+'''''''''''''''
+
+In addition to ``python3`` command changes, there are at least four places in
+CPython that assume the major version is 3 and would need updating:
+
+* `Lib/ast.py <https://github.com/python/cpython/blob/406ffb5293a8c9ca315bf63de1ee36a9b33f9aaf/Lib/ast.py#L50-L51>`__
+* `Parser/pegen.c <https://github.com/python/cpython/blob/406ffb5293a8c9ca315bf63de1ee36a9b33f9aaf/Parser/pegen.c#L654-L658>`__
+* `Parser/pegen.h <https://github.com/python/cpython/blob/406ffb5293a8c9ca315bf63de1ee36a9b33f9aaf/Parser/pegen.h#L284-L288>`__
+* `Parser/string_parser.c <https://github.com/python/cpython/blob/406ffb5293a8c9ca315bf63de1ee36a9b33f9aaf/Parser/string_parser.c#L38-L43>`__
+
+YY.0 rejection
+''''''''''''''
+
+The benefits of calendar versioning are not so big compared to the combined
+costs for YY.0. Therefore, YY.0 is rejected.
+
+YY.MM
+-----
+
+For example, Python 26.10 would be released in October 2026.
+
+Building upon YY.0 we could also include the release month as the minor
+version, like Ubuntu and Black. This makes it clear *when* in the year it was
+released, and also *when* in the year it will reach end-of-life.
+
+However, YY.MM is rejected for many of the same reasons as YY.0.
+
+3.YYYY
+------
+
+For example, Python 3.2026 would be released in 2026.
+
+It's clearer that the minor version is a year when using a four digits, and
+avoids confusion with Ubuntu versions which use YY.MM. However, this is
+rejected as changing from two to four digits would break more code than 3.YY.
+
+Editions
+--------
+
+For example, Python 3.15 (2026 Edition) would be released in 2026.
+
+The Rust language uses
+`"Editions" <https://doc.rust-lang.org/edition-guide/editions/>`__
+to introduce breaking changes. Applying this to Python would require big
+changes to :pep:`387` (Backwards Compatibility Policy) and is out of scope
+for this PEP.
+
+We could apply a year label to releases, such as "Python 3.15 (2026 Edition)",
+but this is rejected because it gives us two numbers to keep track of.
+
+Adopt SemVer and skip 4
+-----------------------
+
+For example, Python 5.0 would be released in 2026, 6.0 in 2027, and so on.
+
+We could skip the problematic 4.0 entirely and adopt SemVer. Because
+deprecations are removed in every feature release, we would get a new major
+bump every year.
+
+This is rejected because we don't get the benefit of calendar versioning, and
+moving away from 3.x would also `break code <PEP 2026 Ecosystem changes_>`_.
+
+Change during 3.14 cycle
+------------------------
+
+The Python 3.14 release must go ahead because: π.
+
+Footnotes
+=========
+
+The author proposed calendar versioning at the `Python Language Summit 2024
+<https://us.pycon.org/2024/events/language-summit/>`__;
+this PEP is a result of discussions there and during PyCon US.
+
+Read the `slides <https://hugovk.github.io/python-calver/>`__
+and TODO `blogpost <TODO>`__ of the summit talk.
```

Comment:
Before merging, I'll add this (if it's been published) or remove this and add it later.

Reply:
Blogposts should be up this week, I'll wait before merging.